examples/samples/partitionedQueues.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  delay,
+  SendableMessageInfo,
+  QueueClient,
+  ReceiveMode,
+  generateUuid,
+  Namespace,
+  ReceiveHandler,
+  OnError,
+  OnMessage,
+  ServiceBusMessage,
+  MessagingError
+} from "../../lib";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const str = process.env.SERVICEBUS_CONNECTION_STRING || "";
+const path = process.env.QUEUE_NAME || "";
+console.log(`str: ${str}`);
+console.log(`path: ${path}`);
+
+let ns: Namespace;
+async function main(): Promise<void> {
+  ns = Namespace.createFromConnectionString(str);
+  const client = ns.createQueueClient(path, { receiveMode: ReceiveMode.peekLock });
+
+  // populate the queue with messages
+  await sendMessages(client);
+
+  let rcvHandler: ReceiveHandler;
+
+  // retrieve messages from the queue
+  const onMessage: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+    console.log(
+      ` \n### Received message:
+        ID - ${brokeredMessage.messageId},
+        messageBody - ${brokeredMessage.body ? brokeredMessage.body.toString() : undefined},
+        partitionKey - ${brokeredMessage.partitionKey},
+        label - ${brokeredMessage.label}`
+    );
+  };
+
+  const onError: OnError = (err: MessagingError | Error) => {
+    console.log(">>>>> Error occurred: ", err);
+  };
+
+  rcvHandler = client.receive(onMessage, onError);
+
+  // wait 5 seconds
+  await delay(5000);
+
+  console.log("Stopping the receiver");
+
+  await rcvHandler.stop();
+
+  console.log("Closing the client");
+
+  await client.close();
+}
+
+async function sendMessages(queueClient: QueueClient): Promise<void> {
+  const data = [
+    { lastName: "Einstein", firstName: "Albert" },
+    { lastName: "Heisenberg", firstName: "Werner" },
+    { lastName: "Curie", firstName: "Marie" },
+    { lastName: "Hawking", firstName: "Steven" },
+    { lastName: "Newton", firstName: "Isaac" },
+    { lastName: "Bohr", firstName: "Niels" },
+    { lastName: "Faraday", firstName: "Michael" },
+    { lastName: "Galilei", firstName: "Galileo" },
+    { lastName: "Kepler", firstName: "Johannes" },
+    { lastName: "Kopernikus", firstName: "Nikolaus" }
+  ];
+
+  for (let j = 0; j < 3; j++) {
```

Comment:
Any idea why we need this outer loop?

Reply:
This would help to show that the messages with same partitionId will go to the same fragment. 
As I mentioned in the below comment(https://github.com/Azure/azure-service-bus-node/pull/123#discussion_r244848283), it makes more sense when we print the SequenceNumber.

Reply:
Currently, the first 2 letters of the name are being used as the partition key. If you use just the first letter, then you can skip the loop. The multiple names starting with the same letter will then have the same partitionId and will go to the same fragment.

To make the sample even more simpler, you can also change your data to be such that few of the names share the same last name. Then you can use the last name as the partition key.

Reply:
I agree, this would significantly reduce the number of messages to send and would simplify the sample.